Mathlib/CategoryTheory/Limits/Shapes/Reflexive.lean

```diff
@@ -148,6 +171,438 @@ instance (priority := 100) hasCoreflexiveEqualizers_of_hasEqualizers [HasEqualiz
+reflexive pair -/
+def reflexivePair (f g : A ⟶ B) (s : B ⟶ A)
+    (sl : s ≫ f = 𝟙 B := by aesop_cat) (sr : s ≫ g = 𝟙 B := by aesop_cat) :
+    (WalkingReflexivePair ⥤ C) where
+  obj x :=
+    match x with
+    | zero => B
+    | one => A
+  map h :=
+    match h with
+    | .id _ => 𝟙 _
+    | .left => f
+    | .right => g
+    | .reflexion => s
+    | .rightCompReflexion => g ≫ s
+    | .leftCompReflexion => f ≫ s
+  map_comp := by
+    rintro _ _ _ ⟨⟩ g <;> cases g <;>
+      simp only [Category.id_comp, Category.comp_id, Category.assoc, sl, sr,
+        reassoc_of% sl, reassoc_of% sr] <;> rfl
+
+section
+
+variable {A B : C}
+variable (f g : A ⟶ B) (s : B ⟶ A) {sl : s ≫ f = 𝟙 B} {sr : s ≫ g = 𝟙 B}
+
+@[simp] lemma reflexivePair_obj_zero : (reflexivePair f g s sl sr).obj zero = B := rfl
+
+@[simp] lemma reflexivePair_obj_one : (reflexivePair f g s sl sr).obj one = A := rfl
+
+@[simp] lemma reflexivePair_map_right : (reflexivePair f g s sl sr).map .left = f := rfl
+
+@[simp] lemma reflexivePair_map_left : (reflexivePair f g s sl sr).map .right = g := rfl
+
+@[simp] lemma reflexivePair_map_reflexion : (reflexivePair f g s sl sr).map .reflexion = s := rfl
+
+end
+
+/-- (Noncomputably) bundle the data of a reflexive pair as a functor out of the walking reflexive
+pair -/
+noncomputable def ofIsReflexivePair (f g : A ⟶ B) [IsReflexivePair f g] :
+    WalkingReflexivePair ⥤ C := reflexivePair f g (commonSection f g)
+
+/-- The natural isomorphism between the diagram obtained by forgetting the reflexion of
+`ofIsReflexivePair f g` and the original parallel pair. -/
+noncomputable def inclusionWalkingReflexivePairOfIsReflexivePairIso
+    (f g : A ⟶ B) [IsReflexivePair f g] :
+    WalkingParallelPair.inclusionWalkingReflexivePair ⋙ (ofIsReflexivePair f g) ≅
+      parallelPair f g :=
+  diagramIsoParallelPair _
+
+end Limits
+
+namespace Limits
+
+variable {C : Type u} [Category.{v} C]
+
+namespace reflexivePair
+
+open WalkingReflexivePair WalkingReflexivePair.Hom
+
+section
+section NatTrans
+
+variable {F G: WalkingReflexivePair ⥤ C}
+/-- A constructor for natural transforms to a diagram of the form `reflexivePair f g s`. -/
+def mkNatTrans (e₀ : F.obj zero ⟶ G.obj zero) (e₁ : F.obj one ⟶ G.obj one)
+    (h₁ : F.map left ≫ e₀ = e₁ ≫ G.map left := by aesop_cat)
+    (h₂ : F.map right ≫ e₀ = e₁ ≫ G.map right := by aesop_cat)
+    (h₃ : F.map reflexion ≫ e₁ = e₀ ≫ G.map reflexion := by aesop_cat) :
+    F ⟶ G where
+  app := fun x ↦ match x with
+    | zero => e₀
+    | one => e₁
+  naturality _ _ f := by
+    cases f
+    all_goals
+      dsimp
+      simp only [Functor.map_id, Category.id_comp, Category.comp_id,
+        Functor.map_comp, h₁, h₂, h₃, reassoc_of% h₁, reassoc_of% h₂,
+        reflexivePair_map_reflexion, reflexivePair_map_left, reflexivePair_map_right,
+        Category.assoc]
+
+@[simp]
+lemma mkNatTrans_app_zero (e₀ : F.obj zero ⟶ G.obj zero) (e₁ : F.obj one ⟶ G.obj one)
```

Comment:
The variables for this lemma and the next are the same as above, so that you should put these three definitions/lemmas in a `section` and use `variable ...`.

Reply:
Is there a syntax to make the autoparam `:= by aesop_cat` apply to only the definition `mkNatTrans` while putting these variables in a `variable ...`? Or should I just leave the autoparam even for the lemmas?